### ISSUE TITLE
Pretraining objectives with MultiLabel?

### ISSUE THREAD

**Issue by ddofer** [UKPLab/sentence-transformers#1320]
Are there any appropriate setups or losses in sentence-transformers for pretraining sentence embeddings in cases where I have labels as targets? 
(I want to finetune the actual embeddings, not just a seperate classifier layer, as I want to analyze the "supervised" embeddings as well. The data is fundamentally that of a single text column, but I can always duplicate part of it to get a 2 sentence format if needed).
Thanks!

**Comment by nreimers** [comment]
Yes:
https://github.com/UKPLab/sentence-transformers/blob/master/examples/training/other/training_batch_hard_trec.py

**Comment by ddofer** [comment]
As written, that assumes multiclass, no? I.E it would make duplicates per label, and in addition you'd need to loop over everything per the number of labels and drop duplicates (input_examples) by the label?

**Comment by nreimers** [comment]
Works for any number of labels. Assumes that you have for every label at least 2 train examples

**Comment by ddofer** [comment]
So, one should use SentenceLabelDataset, with each sentence X label added as a seperate example?
(Won't that mean the same sentence can be a "negative" for itself? e.g. if it appears multiple times for each label, seperately, then it can be a negative for other labels/instances of itself)?

Thanks!

**Comment by nreimers** [comment]
Sentences with the same label are considered as positives, sentences with different labels as negative
